packages/twenty-front/src/modules/workflow/hooks/useCreateNode.tsx

```diff
@@ -0,0 +1,51 @@
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { useUpdateOneRecord } from '@/object-record/hooks/useUpdateOneRecord';
+import {
+  Workflow,
+  WorkflowStep,
+  WorkflowVersion,
+} from '@/workflow/types/Workflow';
+import { getWorkflowLastVersion } from '@/workflow/utils/getWorkflowLastVersion';
+import { insertStep } from '@/workflow/utils/insertStep';
+import { isDefined } from 'twenty-ui';
+
+export const useCreateNode = ({ workflow }: { workflow: Workflow }) => {
+  const { updateOneRecord: updateOneWorkflowVersion } =
+    useUpdateOneRecord<WorkflowVersion>({
+      objectNameSingular: CoreObjectNameSingular.WorkflowVersion,
+    });
+
+  const createNode = ({
+    parentNodeId,
+    nodeToAdd,
+  }: {
+    parentNodeId: string;
+    nodeToAdd: WorkflowStep;
+  }) => {
+    const lastVersion = getWorkflowLastVersion(workflow);
+    if (!isDefined(lastVersion)) {
+      throw new Error(
+        "Can't add a node when no version exists yet. Create a first workflow version before trying to add a node.",
+      );
+    }
+
+    /**
+     * A node id is equal to its related step id.
+     * That's handy when you need to update the steps based on a change on nodes.
+     */
```

Comment:
don't think this comment is useful. This code is explicit enough

Reply:
I will delete the comment.

Choosing to use steps's id as nodes' id wasn't an obvious choice and people who would read the code for the workflows would have to understand that I simplified the codebase that way.

I usually maintain a document explaining all my technical choices. It makes it easy to understand why a choice was made. I don't know if you have something similar but that's where I would put this comment.